quickfixj-core/src/main/java/quickfix/Message.java

```diff
@@ -377,6 +377,8 @@ public String toXML(DataDictionary dataDictionary, boolean indent) {
             if (indent) {
                 serializer.setOutputProperty(OutputKeys.INDENT, "yes");
                 serializer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "4");
+            } else {
+                serializer.setOutputProperty(OutputKeys.INDENT, "no");
```

Comment:
I know it does not make any sense, but to be backward compatible we need to set `OutputKeys.INDENT, "yes"` without setting the `indent-amount`, right?

Reply:
In theory. If we do that then we are backwards compatible, but we get different behaviour in Java 11. If we set to "no" we are backwards compatible and consistent across Java versions.

BTW. Fixing another problem. Seems that there was a bug in Java 9/10/11 https://bugs.openjdk.java.net/browse/JDK-8215543 that also puts additional new line characters before and after `CDATA` elements when indent is enabled.